src/SecretsFile.hs

```diff
@@ -0,0 +1,166 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module SecretsFile where
+
+import Control.Applicative.Combinators (some, option, optional, sepBy1)
+import Control.Exception (try)
+import Control.Monad.Except (MonadError, MonadIO, liftEither, liftIO)
+import Data.Char (toUpper, isSpace)
+import Data.Functor (void)
+import Data.List (intercalate)
+import Data.Void (Void)
+import qualified Text.Megaparsec as MP
+import qualified Text.Megaparsec.Char as MPC
+import qualified Text.Megaparsec.Char.Lexer as MPL
+
+data Secret = Secret
+  { sMount :: String
+  , sPath :: String
+  , sKey :: String
+  , sVarName :: String
+  } deriving (Eq, Show)
+
+data SFVersion
+  = V1
+  | V2
+  deriving (Show)
+
+type Parser = MP.Parsec Void String
+
+data SFError = IOError | ParseError (MP.ParseError (MP.Token String) Void)
```

Comment:
Actually, a more important point:
You are shadowing the built-in `type IOError = GHC.IO.Exception.IOException` here.
That is pretty confusing (esp. since you are also using that type further down in `Either IOError String`), so maybe better call it `SomeIOError` or something like that.

Reply:
This was actually pretty easy thanks to `displayException`